pkg/controllers/termination/suite_test.go

```diff
@@ -116,29 +127,68 @@ var _ = Describe("Termination", func() {
 			node = ExpectNodeExists(env.Client, node.Name)
 			ExpectReconcileSucceeded(controller, client.ObjectKeyFromObject(node))
 
+			// Expect no pod to be enqueued for eviction
+			ExpectPodsNotEnqueuedForEviction(env.Client, enqueued, podEvict)
+			ExpectPodsNotEnqueuedForEviction(env.Client, enqueued, podNoEvict)
+
 			// Expect node to exist, but be cordoned
 			node = ExpectNodeExists(env.Client, node.Name)
-			Expect(node.Spec.Unschedulable).To(Equal(true))
-
-			// Expect pods to not be evicting
-			podEvict = ExpectPodExists(env.Client, podEvict.Name, podEvict.Namespace)
-			Expect(podEvict.GetObjectMeta().GetDeletionTimestamp().IsZero()).To(BeTrue())
-			podNoEvict = ExpectPodExists(env.Client, podNoEvict.Name, podNoEvict.Namespace)
-			Expect(podNoEvict.GetObjectMeta().GetDeletionTimestamp().IsZero()).To(BeTrue())
+			Expect(node.Spec.Unschedulable).To(BeTrue())
 
 			// Delete do-not-evict pod
 			ExpectDeleted(env.Client, podNoEvict)
 
 			// Reconcile node to evict pod
 			node = ExpectNodeExists(env.Client, node.Name)
 			ExpectReconcileSucceeded(controller, client.ObjectKeyFromObject(node))
-			pod := ExpectPodExists(env.Client, podEvict.Name, podEvict.Namespace)
-			Expect(pod.GetObjectMeta().GetDeletionTimestamp().IsZero()).To(BeFalse())
+
+			// Expect podEvict to be enqueued for eviction then be successful
+			ExpectPodsEnqueuedForEviction(env.Client, enqueued, podEvict)
+			ExpectPodsEvictingSucceeded(env.Client, podEvict)
 
 			// Delete pod to simulate successful eviction
-			ExpectDeleted(env.Client, pod)
+			ExpectDeleted(env.Client, podEvict)
 
-			// Terminate Node
+			// Reconcile to delete node
+			node = ExpectNodeExists(env.Client, node.Name)
+			ExpectReconcileSucceeded(controller, client.ObjectKeyFromObject(node))
+			ExpectNotFound(env.Client, node)
+		})
+		It("should fail to evict pods that violate a PDB", func() {
+			key, value := randomdata.SillyName(), randomdata.SillyName()
+			pdb := test.PodDisruptionBudget(test.PDBOptions{
```

Comment:
What do you think about simplifying the options struct to avoid leaking some of this type conversion, e.g.
```
test.PDBOptions{
  LabelSelector: map[string]string{key: value},
  MinAvailable: 1,
}
```
Do the type conversion inside your helper.

Reply:
It's a little different in that PDB options should be able to specify intstr.IntorString, but the default value of the non-pointer value is 0, which is still a valid PDB option. Since you can't specify both maxUnavailable and minAvailable, we then run into a PDB configuration issue.